### PR TITLE
Implementation of  digitizer noise occupancy 

### PR DESCRIPTION
**Pull request by suchandradutta**
#### PR description:
 Implementation of noise occupancy in neighboring channels without containing signal Hits for strip like modules (PSS,SS) in the Phase2 Outer Tracker. The threshold cutoff for PSS and SS modules are updated to 4800e and 6000e respectively. Corresponding noise values are set as 1010e and 1263e considering the threshold to be 4.75sigmas. The add_noisy_cell method has been updated to use this cutoff. Other than this change, 
   - we have switched off threshold smearing in all types of modules. 
   - at the moment the configuration parameter ReadoutNoiseInElec is not used and kept with a dummy value for the time being

#### PR validation:

 The performance were presented in Phase2 Tracker Simulation meetings 
       https://indico.cern.ch/event/1201188/contributions/5095554/attachments/2528800/4350459/Phase2Digitizer_Noise_occupancy_14102022.pdf

https://indico.cern.ch/event/1201186/contributions/5051043/attachments/2510200/4314408/Phase2Digitizer_Noise_occupancy_16092022.pdf

### CODE REVIEW

File: SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc [hunk 1]
```diff
@@ -891,8 +891,12 @@ void Phase2TrackerDigitizerAlgorithm::digitize(const Phase2TrackerGeomDetUnit* p
     add_noise(pixdet);  // generate noise
   if (addXtalk_)
     add_cross_talk(pixdet);
-  if (addNoisyPixels_)
-    add_noisy_cells(pixdet, theHIPThresholdInE / theElectronPerADC_);
+  if (addNoisyPixels_) {
+    float thresholdInNoiseUnits = 99.9;
```
Comment (civanch) [inline SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc:895]: @suchandradutta , should this value be in the list of parameters?, is it const or may be era dependent?

**Comment by civanch** [comment]
@suchandradutta , PR needs some description.

File: SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc (hunk 1 above)
Comment (suchandradutta) [reply SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc:895]: @civanch this is not era dependent. This a very large sigma value (somewhat arbitrary) to protect the add_noisy_cells function when the channel noise is zero.

**Comment by suchandradutta** [comment]
> @suchandradutta , PR needs some description.
Sorry! I did add the comment but somehow it did not show up. Added it again now

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

there is some changes in upgrade WFs , which should be expected if the noise sampling is changed.

**Comment by srimanob** [comment]
+Upgrade

Note to @cms-sw/pdmv-l2, we should see its effect in the validation.

**Comment by rappoccio** [comment]
+1

Minimal code changes, mostly updates to constants, PDMV should keep an eye on these in validation.

**Pull request merged**